projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -270,6 +270,52 @@ protected function get_cast_option_value_or_null( $option_name, $cast_callable )
 		return call_user_func( $cast_callable, $option_value );
 	}
 
+	/**
+	 * Used for the WooCommerce address options. This function returns the option value if it exists, else it looks
+	 * for a default address on the domain contact information. If that does not exist it returns an empty string.
+	 *
+	 * @param string $key The option key to retrieve a value.
+	 *
+	 * @return string The requested address part value or an empty string.
+	 */
+	protected function get_woocommerce_address( $key ) {
+		$address_part = get_option( $key );
+
+		// If there's any value set in the database, return it. Even if it's null or an empty string.
+		// False means no value for this option was found in the database.
+		if ( false !== $address_part ) {
+			return $address_part;
+		}
+
+		// If no value is set in the database, return the corresponding address part from the user's domain contact
+		// address. If that does not exist return an empty string.
+		$address_part = '';
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$address = Domain_Contact_Information_Mapper::find_by_user_id( get_current_user_id() );
```

Comment:
I don't think prefill logic should be happening here. This endpoint should only return site setting values as they are.

Falling back on other data means you can't tell if its a site setting or if its fallback info. It works for our case because we want to prefill the data and let the user update it if it's wrong but if something else requests the data it can't trust if its the site setting is or some prefill data based on the current user making the request.

If we want to prefill form data we should fetch the users contact info separately and decide what to do in the front end.

I'm also wondering if prefilling by default is even the right call, there are a few situations where you want to list a business address different from your domain registration address, it might make more sense to provide a nudge to trigger prefilling with a "known address" rather than prefilling by default (@SaxonF thoughts?)

Maybe we skip prefill for this sprint and add it as an optimization later? Definitely like the idea having a "prefil with address 123 street..." nudge on the form.

Reply:
> Falling back on other data means you can't tell if its a site setting or if its fallback info.

This is a great point. Additionally, the logic for formatting the domain information for our purposes is growing in complexity, and might be better suited outside of this endpoint. (Though I just checked in my latest updates for posterity.)

Reply:
Removed this feature code from this PR and added a draft issue to follow up on in the backlog.